src/pages/index.js

```diff
@@ -138,36 +138,39 @@ function createCard(data) {
 
 function handleProfileEditSubmit(data) {
   api.updateProfile(data.name, data.profession)
-  .then((data) => {
-    userInfo.setUserInfo(data)
-  })
-  .catch((err) => {
-    console.log(`An error occured: ${err}`);
-  });
+    .then((data) => {
+      userInfo.setUserInfo(data)
+    })
+    .catch((err) => {
+      console.log(console.error);
+    });
   editProfilePopup.close();
-}  
+}
 
 function handleAvatarChange(data) {
   api.updateAvatar(data)
-  .then((data) => {
-  userInfo.setUserInfo(data)
-  })
-  .catch((err) => {
-    console.log(`An error occured: ${err}`);
-  });
-  editProfilePopup.close();
-}  
+    .then((data) => {
+      avatarPopup.renderLoading(false)
+      userInfo.setUserInfo(data)
+    })
+    .then(avatarPopup.close())
+    .finally(avatarPopup.renderLoading(true))
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.